synapse/lib/crypto/coin.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import binascii
+
+import cbor  # cardano
+import regex
+import bech32  # cardano
+import base58  # btc and cardano
+import _pysha3  # do not import the sha3 library directly.
+import bitcoin  # btc
+import bitcoin.bech32 as bitcoin_b32  # btc
+import cashaddress.convert as cashaddr_convert  # BCH support
+import xrpl.core.addresscodec as xrp_addresscodec  # xrp support
+import substrateinterface.utils.ss58 as substrate_ss58  # polkadot support
+
+import synapse.common as s_common
+
+logger = logging.getLogger(__name__)
+
+'''
+synapse.lib.crypto.coin contains functions for verifying whether or not a given
+regex match containing a valu is valid for a given type of coin.
+
+these functions are intended to be used with synapse.lib.scrape.
+'''
```

Comment:
We can move these functions back into scrape if we want to do so; I placed them here at least for keeping the underlying changes to scrape.py a bit cleaner.

Reply:
Seems reasonable to have them outside of scrape